### PR TITLE
Extract providers

### LINKED ISSUE #1873: Clean out Velero of cloud provider code
This is to track the removal of the provider code. Other issues will track the implementation of each as a separate project.

### PR DESCRIPTION
**Pull request by carlisia**
Closes #1873.

Take II of https://github.com/vmware-tanzu/velero/pull/1890, now with dependency mess fixed.

TODOs for this PR:
- [x] Remove provider documentation
- [x] Add links to external repos for provider documentation
- [ ] Ensure we have clean documentation for how to install any external plugins
- [x] Resolve the issue of restic needing Azure variables/configuration info
- [x] Clean up dependencies

Not planned TODO for this PR:
- 🚫 Remove AWS dependency for restic config

Before merging this PR:
- [x] get the new repos setup
- [x] get image builds for the new repos set up
- [x] make `install plugins` required

### CODE REVIEW

**Comment by carlisia** [comment]
Sprucing up docs next (tomorrow). Maybe I'll make the `install plugins` required in a separate PR.

File: pkg/persistence/in_memory_object_store.go [hunk 1]
```diff
@@ -1,20 +1,17 @@
 /*
 Copyright 2018 the Velero contributors.
-
```
Comment (skriss) [inline pkg/persistence/in_memory_object_store.go]: let's not remove this or the following 2 blank lines from the copyright header

File: pkg/cmd/cli/install/install.go [hunk 2]
```diff
@@ -347,5 +347,9 @@ func (o *InstallOptions) Validate(c *cobra.Command, args []string, f client.Fact
 		return errors.New("--default-restic-prune-frequency must be non-negative")
 	}
 
+	if len(o.Plugins) == 0 {
```
Comment (skriss) [inline pkg/cmd/cli/install/install.go]: I'm wondering if we should make this flag optional in the case where the user is not configuring a BSL or a VSL (i.e. L333 - https://github.com/vmware-tanzu/velero/pull/1985/files/f8b1e46469794dbc0250d4ddb9887d3760c7fcf2..006e1ca8b9b546e2ceeaeee8a352cdd351afcf00#diff-6dea8f0eef5112cb00bdce50e2cb339cR333).

Comment (carlisia) [reply pkg/cmd/cli/install/install.go]: Yes, I guess so. If those are going to be set later, then we can let them add the plugins later too.

Comment (skriss) [reply pkg/cmd/cli/install/install.go]: Yeah, I think that makes sense. Let's make that change - then I think this is good to go!

Comment (carlisia) [reply pkg/cmd/cli/install/install.go]: Done!

**Comment by skriss** [comment]
@prydonius PTAL

**Comment by carlisia** [comment]
> Deep breath...remove 250k lines of code... 😄

Most of which... dependencies!

**Pull request merged**